impl/src/main/resources/migrations.xml

```diff
@@ -206,4 +206,88 @@
         <dropColumn tableName="question" columnName="votes"/>
     </changeSet>
 
+    <changeSet id="tagging" author="tobias.friberg">
+        <createTable tableName="tag">
+            <column autoIncrement="true" name="id" type="SERIAL">
+                <constraints primaryKey="true" primaryKeyName="tag_pk" nullable="false"/>
+            </column>
+
+            <column name="label" type="VARCHAR">
+                <constraints unique="true" nullable="false"/>
+            </column>
+        </createTable>
+
+        <createTable tableName="question_tag">
+            <column name="question_id" type="INTEGER">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+
+            <column name="tag_id" type="INTEGER">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+        </createTable>
+        <addForeignKeyConstraint baseColumnNames="question_id"
+                                 baseTableName="question_tag"
+                                 constraintName="question_tag_question_fk"
+                                 onDelete="CASCADE"
+                                 onUpdate="NO ACTION"
+                                 referencedColumnNames="id"
+                                 referencedTableName="question"/>
+
+        <addForeignKeyConstraint baseColumnNames="tag_id"
+                                 baseTableName="question_tag"
+                                 constraintName="question_tag_tag_fk"
+                                 onDelete="CASCADE"
+                                 onUpdate="NO ACTION"
+                                 referencedColumnNames="id"
+                                 referencedTableName="tag"/>
+    </changeSet>
+
+    <changeSet id="clean-up-dangling-tags-by-trigger" author="magnus.persson">
```

Comment:
I'm not really a fan of using logic in the database like this. It could also introduce weird upgrade issues like keeping backwards compatibility while/before the backends are upgrading.
Isn't keeping dangling tags in order just as simple (or simpler) to do in the code where updates on tag or deletions are made?

Reply:
Will use DaoTransactions instead. The trigger was used to let postgres be responsible handling unused tags when removing associations on the question_tag table instead of having developers remember this :)